ssc/cmod_wavefile.cpp

```diff
@@ -60,100 +78,255 @@ class cm_wave_file_reader : public compute_module
+                    
+                }
+            }
+            ssc_number_t hour0, hour1, hourdiff;
+            ssc_number_t ts_significant_wave_height;
+            ssc_number_t ts_energy_period;
+            size_t ncols = 22;
+            size_t nrows = 21;
+            size_t sig_wave_height_index = 0;
+            size_t energy_period_index = 0;
+            ssc_number_t* p_year = allocate("year", numberRecords);
+            ssc_number_t* p_day = allocate("day", numberRecords);
+            ssc_number_t* p_hour = allocate("hour", numberRecords);
+            ssc_number_t* p_minute = allocate("minute", numberRecords);
+            ssc_number_t* mat = allocate("wave_resource_matrix", nrows, ncols);
+            //size_t numberRecords = wave_dp->nrecords();
+            ssc_number_t* month = allocate("month", numberRecords);
+            std::vector<ssc_number_t> timecheck(numberRecords);
+            timecheck[0] = 0;
+            ssc_number_t* wave_heights = allocate("wave_significant_height", numberRecords);
+            ssc_number_t* wave_periods = allocate("wave_energy_period", numberRecords);
+            for (size_t r = 0; r < numberRecords; r++) {
+                getline(ifs, buf);
+                values.clear();
+                values = split(buf);
+                
+                if (r == 0) {
+                    //value_0 = split(buf);
+                    hour0 = (ssc_number_t)std::stod(values[hour_index]);
+                }
+                if (r == 1) {
+                    //value_1 = split(buf);
+                    hour1 = (ssc_number_t)std::stod(values[hour_index]);
+                    hourdiff = hour1 - hour0;
+                }
+                month[r] = (ssc_number_t)std::stod(values[month_index]);
+                timecheck[r] = (ssc_number_t)std::stod(values[hour_index]);
+                if (r > 0) {
+                    if (timecheck[r] - timecheck[r - 1] != hourdiff && timecheck[r] != 0) {
+                        throw exec_error("wave_file_reader", "Time steps are nonuniform");
+                        timecheck[r] = 999;
+                    }
+                }
+                p_year[r] = (ssc_number_t)std::stod(values[year_index]);
+                p_hour[r] = (ssc_number_t)std::stod(values[hour_index]);
+                p_day[r] = (ssc_number_t)std::stod(values[day_index]);
+                p_minute[r] = (ssc_number_t)std::stod(values[minute_index]);
+                wave_heights[r] = (ssc_number_t)std::stod(values[height_index]);
+                wave_periods[r] = (ssc_number_t)std::stod(values[period_index]);
+
+                //Make JPD from time series data
+
+                ts_significant_wave_height = wave_heights[r];
+                ts_energy_period = wave_periods[r];
+                for (size_t j = 0; j < 21; j++) {
+
+                    if (r == 0) mat[(j + 1) * ncols] = (0.25 + j * 0.5);
```

Comment:
move this and line 273 to right under the instantiation of `mat` in line 224, since this appears to be initializing the bin widths of matrix. That would also be a good place to add in a comment that the rows are for height and columns are for period